src/Adapter/Product/CommandHandler/UpdateProductOptionsHandler.php

```diff
@@ -0,0 +1,259 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductHandler;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductOptionsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\CommandHandler\UpdateProductOptionsHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShopException;
+use Product;
+use Tag;
+use Validate;
+
+/**
+ * Handles UpdateProductOptionsCommand using legacy object models
+ */
+final class UpdateProductOptionsHandler extends AbstractProductHandler implements UpdateProductOptionsHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateProductOptionsCommand $command): void
+    {
+        $product = $this->getProduct($command->getProductId());
+        $this->fillUpdatableFieldsWithCommandData($product, $command);
+        $product->setFieldsToUpdate($this->fieldsToUpdate);
+
+        $this->performUpdate($product, $command->getLocalizedTags());
+    }
+
+    /**
+     * @param Product $product
+     * @param UpdateProductOptionsCommand $command
+     */
+    private function fillUpdatableFieldsWithCommandData(Product $product, UpdateProductOptionsCommand $command): void
+    {
+        if (null !== $command->getVisibility()) {
+            $product->visibility = $command->getVisibility()->getValue();
```

Comment:
Or do we consider they are useless because the ValueObjects handle this? Though if someone overrides Product class (or we update it ourselves) the new validation rules will be ignored
I wonder if it's better to have a double check 🤔

Reply:
yes I've removed it because it does the same.
And It is our domain VO it cannot be overriden by something else, so we shouldn't care about legacy validation anymore.